server/src/test/java/org/elasticsearch/gateway/Zen2GatewayMetaStateTests.java

```diff
@@ -0,0 +1,235 @@
+                        .build()
+        ).version(version).build();
+    }
+
+    public void testSetLastAcceptedState() throws IOException {
+        Gateway gateway = newGateway();
+        final long term = randomNonNegativeLong();
+
+        for (int i = 0; i < randomIntBetween(1, 5); i++) {
+            final long version = randomNonNegativeLong();
+            final String indexName = randomAlphaOfLength(10);
+            final IndexMetaData indexMetaData = createIndexMetaData(indexName, randomIntBetween(1,5), randomNonNegativeLong());
+            final MetaData metaData = MetaData.builder().
+                    persistentSettings(Settings.builder().put(randomAlphaOfLength(10), randomAlphaOfLength(10)).build()).
+                    coordinationMetaData(createCoordinationMetaData(term)).
+                    put(indexMetaData, false).
+                    build();
+            ClusterState state = createClusterState(version, metaData);
+
+            gateway.setLastAcceptedState(state);
+            gateway = maybeNew(gateway);
+            assertThat(gateway.getLastAcceptedState().version(), equalTo(version));
+            assertTrue(MetaData.isGlobalStateEquals(gateway.getLastAcceptedState().metaData(), metaData));
+            assertThat(gateway.getLastAcceptedState().metaData().index(indexName), equalTo(indexMetaData));
+        }
+    }
+
+    public void testSetLastAcceptedStateTermChanged() throws IOException {
+        Gateway gateway = newGateway();
+
+        final String indexName = randomAlphaOfLength(10);
+        final int numberOfShards = randomIntBetween(1, 5);
+        final long version = randomNonNegativeLong();
+        final long term = randomNonNegativeLong();
+        final IndexMetaData indexMetaData = createIndexMetaData(indexName, numberOfShards, version);
+        final ClusterState state = createClusterState(randomNonNegativeLong(),
+                MetaData.builder().coordinationMetaData(createCoordinationMetaData(term)).put(indexMetaData, false).build());
+        gateway.setLastAcceptedState(state);
+
+        gateway = maybeNew(gateway);
+        final long newTerm = randomValueOtherThan(term, () -> randomNonNegativeLong());
+        final int newNumberOfShards = randomValueOtherThan(numberOfShards, () -> randomIntBetween(1,5));
+        final IndexMetaData newIndexMetaData = createIndexMetaData(indexName, newNumberOfShards, version);
+        final ClusterState newClusterState = createClusterState(randomNonNegativeLong(),
+                MetaData.builder().coordinationMetaData(createCoordinationMetaData(newTerm)).put(newIndexMetaData, false).build());
+        gateway.setLastAcceptedState(newClusterState);
+
+        gateway = maybeNew(gateway);
+        assertThat(gateway.getLastAcceptedState().metaData().index(indexName), equalTo(newIndexMetaData));
+    }
+
+    public void testCurrentTermAndTermAreDifferent() throws IOException {
```

Comment:
what's the benefit of this test? By asserting what's not changed in the above tests. I think that it also covers this test?

Reply:
Replying to this and two comments above. I just want to make `testSetCurrentTerm` to test only `seCurrentTerm` and `testSetLastAcceptedState` to test only `setLastAcceptedState`. As you may notice `testSetCurrentTerm` and `testSetLastAccepted` tests do not event call `setLastAcceptedState` and `setCurrentTerm`, they focus exclusively on testing their methods. This test focuses on these two methods interaction because it's explicitly required by JavaDoc.